pkg/utils/nodepool/nodepool.go

```diff
@@ -39,6 +39,15 @@ func IsManaged(nodePool *v1.NodePool, cp cloudprovider.CloudProvider) bool {
 	})
 }
 
+func GetNodeClassStatusObject(nodePool *v1.NodePool, cp cloudprovider.CloudProvider) status.Object {
```

Comment:
I would just call it `GetNodeClass()` -- yeah, the thing that you are returning is `status.Object`, but that's just because all NodeClasses have to implement this interface